ApmServer_should_accept_event_and_write_data_to_Elasticsearch fails in 8.9.0-SNAPSHOT

[cloud-on-k8s-operator-nightly/builds/217@stack-8.9.0-SNAPSHOT](https://buildkite.com/elastic/cloud-on-k8s-operator-nightly/builds/217#0188dfec-e18a-4ab9-b938-04ce543b4953)

- TestCrossNSAssociation/ApmServer_should_accept_event_and_write_data_to_Elasticsearch
- TestUpdateConfiguration/ApmServer_should_accept_event_and_write_data_to_Elasticsearch
- TestAPMAssociationWhenReferencedESDisappears/ApmServer_should_accept_event_and_write_data_to_Elasticsearch


```sh
--- FAIL: TestCrossNSAssociation (1893.90s)
    --- FAIL: TestCrossNSAssociation/ApmServer_should_accept_event_and_write_data_to_Elasticsearch (1800.00s)
                Error Trace:    /go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/utils.go:94
                Error:          Received unexpected error:
                                1 documents expected in index logs-apm.error-default, got 0 instead

--- FAIL: TestAPMAssociationWhenReferencedESDisappears (1872.86s)\n"
    --- FAIL: TestAPMAssociationWhenReferencedESDisappears/ApmServer_should_accept_event_and_write_data_to_Elasticsearch (1800.00s)\n"
                Error Trace:    /go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/utils.go:94
                Error:          Received unexpected error:
                                1 documents expected in index logs-apm.error-default, got 0 instead

--- FAIL: TestUpdateConfiguration (1896.64s)
    --- FAIL: TestUpdateConfiguration/ApmServer_should_accept_event_and_write_data_to_Elasticsearch (1800.00s)
                Error Trace:    /go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/utils.go:94
                Error:          Received unexpected error:
                                1 documents expected in index logs-apm.error-default, got 0 instead
```

--- a/test/e2e/test/apmserver/builder.go
+++ b/test/e2e/test/apmserver/builder.go
@@ -19,8 +19,6 @@ import (
 	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test"
 )
 
-const APMServerDataStreamsWaitForIntegration = "apm-server.data_streams.wait_for_integration"
-
 // Builder to create APM Servers
 type Builder struct {
 	ApmServer apmv1.ApmServer
@@ -32,11 +30,6 @@ var _ test.Builder = Builder{}
 var _ test.Subject = Builder{}
 
 func (b Builder) SkipTest() bool {
-	// https://github.com/elastic/cloud-on-k8s/issues/6947
-	if test.Ctx().ElasticStackVersion == "8.9.0-SNAPSHOT" && !b.HasIntegration() {
-		return true
-	}
-
 	// APM doesn't work in 8.5.3, see https://github.com/elastic/apm-server/issues/10089.
 	return test.Ctx().ElasticStackVersion == "8.5.3"
 }
@@ -190,26 +183,8 @@ func (b Builder) WithoutIntegrationCheck() Builder {
 	}
 
 	return b.WithConfig(map[string]interface{}{
-		APMServerDataStreamsWaitForIntegration: false,
+		"apm-server.data_streams.wait_for_integration": false,
 	})
-}
-
-// HasIntegration returns true if the APM Server uses the APM integration through Kibana to setup the index templates.
-// The detection is based on a setting that we set to disable the integration check.
-func (b Builder) HasIntegration() bool {
-	if b.ApmServer.Spec.Config != nil || b.ApmServer.Spec.Config.Data != nil {
-		v, ok := b.ApmServer.Spec.Config.Data[APMServerDataStreamsWaitForIntegration]
-		if !ok {
-			// integration is used if no setting
-			return true
-		}
-		if value, ok := v.(bool); ok {
-			// integration if the setting value is true
-			return value
-		}
-		return ok
-	}
-	return true
 }
 
 func (b Builder) WithMutatedFrom(builder *Builder) Builder {

